pkg/utils/provider_meta.go

```diff
@@ -0,0 +1,94 @@
+package utils
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/MaterializeInc/terraform-provider-materialize/pkg/clients"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jmoiron/sqlx"
+)
+
+type ProviderMeta struct {
```

Comment:
This struct seems important. Maybe add some additional comments?

Reply:
Good point! Just added some comments.